src/components/welcome/MyAccounts/index.tsx

```diff
@@ -51,6 +61,13 @@ const AccountsList = ({ safes, onLinkClick }: AccountsListProps) => {
     ],
     [safes, watchlistMultiChainSafes],
   )
+  const pinnedSafes = useMemo<(MultiChainSafeItem | SafeItem)[]>(
+    () => [
+      ...(safes.allSingleSafes?.filter(({ isPinned }) => isPinned) ?? []),
+      ...(safes.allMultiChainSafes?.filter(({ isPinned }) => isPinned) ?? []),
+    ],
+    [safes],
+  )
```

Comment:
I am getting a flicker in the accounts list when loading the app. It looks like `pinnedSafes` are there but then for a few render cycles it is an empty array again and then switches back to the correct result. Is there a way we can pass in a loading flag from the `useAllSafes` hook and wait until everything is ready before rendering the list?